serverlessworkflow/modules/ROOT/pages/eventing/working-with-openapi-callbacks.adoc

```diff
@@ -0,0 +1,121 @@
+= OpenAPI Callback in {context}
+
+:compat-mode!:
+// Metadata:
+:description: Working with openapi callbacks
+:keywords: kogito, workflow, serverless, openapi, callback, event
+
+This document describes the link:{open_api_spec_url}#callback-object[Openapi Callback] and related examples. Also, it covers the difference between OpenAPI Callback and the Serverless Workflow Callback state. Both the OpenAPI callback and the Callback state perform an action and wait for an event to be produced as a result of the action to resume the workflow. Therefore, both OpenAPI callback and the Callback state are suitable to perform  _fire & wait-for-result_ operations. 
+
+The OpenAPI callback is an asynchronous, out-of-band request that your service will send to some other service in response to specific events. In contrast, the action performed by a Callback state is an asynchronous external service invocation. Both ways aim to send an event to the workflow to resume the flow.
+
+As mentioned in the link:{open_api_spec_url}#callback-object[OpenAPI Callback document], when the workflow makes an asynchronous service call, you must provide the callback URL in the request body. The external service invokes the callback URL and sends a link:{cloud_events_url}[CloudEvent] to the workflow.
+
+From a workflow perspective, asynchronous service indicates that the control is returned to the caller immediately without waiting for the action to be completed. Once the action is completed, a link:{cloud_events_url}[CloudEvent] needs to be published to resume the workflow.
+
+For the workflow to identify the published CloudEvent it is waiting for, the external service developer includes the workflow instance ID in the CloudEvent header or uses the xref:eventing/event-correlation-with-workflows.adoc[Event correlation]. The following figure displays the process:
+
+image::eventing/callbacks-explained.png[]
+
+The workflow correlation described in this document focuses on the former mechanism that is based on the fact that each workflow instance contains a unique identifier, which is generated automatically.
+
+[[ref-example-callback]]
+== Example of the OpenAPI Callback
+
+To understand the OpenAPI Callback, see the link:{kogito_sw_examples_url}/serverless-workflow-callback-events-over-http-quarkus[`serverless-workflow-callback-events-over-http-quarkus`] example application in GitHub repository.
+
+The `serverless-workflow-callback-events-over-http-quarkus` application is initiated with the following request:
+[code,json]
+----
```

Comment:
Which URI should it be sent to ?

Reply:
The URI and other details to run the example were mentioned in the example repo- https://github.com/kiegroup/kogito-examples/tree/stable/serverless-workflow-examples/serverless-workflow-callback-events-over-http-quarkus. The same has been mentioned in the document.